packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal.sol

```diff
@@ -116,7 +125,7 @@ contract FuelMessagePortal is
 
     /// @notice Contract initializer to setup starting values
     /// @param fuelChainState Chain state contract
-    function initialize(FuelChainState fuelChainState) public initializer {
+    function initialize(FuelChainState fuelChainState, uint256 _limitAmount, uint256 _rateLimitDuration) public initializer {
```

Comment:
Make this function virtual and override it in V3. Add the new variables there. FuelMessagePortal.sol should remain unmodified

Reply:
thanks wasn't aware that it should be immutable :)